### ISSUE TITLE
Possibly Improved Sequence Tagger

### ISSUE THREAD

**Issue by ghost** [NYTimes/ingredient-phrase-tagger#9]
Hi,
     This is a pretty awesome project, thanks for posting it! I've been experimenting with Structured Prediction methods, and decided to use this project to compare CRFs and Learning to Search methods. Pending a more rigorous evaluation (_fingers crossed_) I'm seeing roughly 96% per-token accuracy and 95.55% sentence-level accuracy with L2S and [vw](https://github.com/JohnLangford/vowpal_wabbit/wiki), taking 22 minutes total for read+train+test. This is an 80/20 split on the full dataset, using the output of `bin/generate`. Once I've cleaned up the source, I'll be happy to send over a pull request.

\- Arthur

**Comment by tettoffensive** [comment]
@Zintinio did you ever clean up the source? I'm curious to see your improvements. I'm seeing a few things where I'll get "name": "Salt and pepper" instead of two separate ingredients. Wondering if your improvements would help with this sort of problem?

Also, ingredients like "Basil" and "Basil leaves". I think it would be better if they were both recognized as the same ingredient. But that might be much more challenging ;)

**Comment by ghost** [comment]
Yeah I'll dig it up. It's actually just using Vowpal Wabbit + Dagger, with
the same features.
